src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/DependencyInjectionWithLegacyFallbackInstanceProvider.cs

```diff
@@ -3,66 +3,127 @@
 
 using System;
 using CoreWCF.Channels;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace CoreWCF.Dispatcher
 {
     internal class DependencyInjectionWithLegacyFallbackInstanceProvider : IInstanceProvider
     {
+        class ScopedServiceProviderExtension : IExtension<InstanceContext>, IServiceProvider, IDisposable
```

Comment:
I just noticed this is missing an accessor. I think it defaults to private as it's an inner class. Can you make this explicit.

Reply:
done!